### ISSUE TITLE
$('div.display-logic').getFieldName() broken with SilverStripe >=3.2

### ISSUE THREAD

**Issue by Zauberfisch** [unclecheese/silverstripe-display-logic#36]
In SilverStripe it was usual that FormFields had the ID set to the name of the FormField. With SilverStripe 3.2 FormFields now have a 'namespaced' ID.

`<div class="field text" id="FirstName">` becomes `<div class="field text" id="FormClass_FormName_FirstName_Holder">`

but the name of the input remains the same (`<input name="FirstName">`).
because of this, display_logic is no longer able to find the input.

---

I have hot-fixed the problem for myself now overwriting `getFieldName` with this in my projects javascript file:

    $('form div.display-logic, form div.display-logic-master').entwine({
		getFieldName: function () {
			return this.attr('id').replace(new RegExp('^' + this.closest('form').attr('id') + '_(.*)_Holder$'), '$1');
		}
	});

**Comment by jeffwhitfield** [comment]
Started testing the latest 3.2 release candidate (v3.2.0-rc1). Unfortunately, the display logic module is dead on arrival. I'm assuming this particular bug is the reason?

**Comment by Zauberfisch** [comment]
probably, though entwike allows you to overwrite the getFieldName() method, so you could hotfix it yourself I suppose

**Comment by jeffwhitfield** [comment]
Yeah, unfortunately, your hotfix doesn't appear to be working at the moment. Could be that the fieldname convention changed with later betas. :P

**Comment by mediabeastnz** [comment]
Doesn't work with stable 3.2 either ;)

**Comment by Oliver996** [comment]
I have fixed the problem on my fork. I changed how the master is set from the fieldname to a extraclass on the masterfield and together with the fix from Zauberfisch it works now on 3.2:
 
```php
$fields->addFieldsToTab(
    'Root.Main',
    array(
       CheckboxField::create('Exampelmaster')->addExtraClass('exampel-class'),
       TextField::create('Textfield')->displayIf('exampel-class')->isChecked()->end(),
    )
);
```

It is the same problem as Zauberfisch already described at the beginning. The master has also the problem with the new ID. Then you just set the fieldname for the master `->displayif('fieldname')`, but now the master needs the full ID "FormClass_FormName_fieldname_Holder". I was able to get the FieldID(fieldname_Holder), but not the current Name of the form. Thats why I changed it to a class.

The better solution would be to create the full field ID("FormClass_FormName_fieldname_Holder"). It's not fully tested, but for me the fix works.

**Comment by jeffwhitfield** [comment]
I'll play around with this fix and see if it does the trick. If so, we should probably create a pull request for the fix. This particular add-on is used quite a bit in a few other add-ons. Without the fix, upgrades to 3.2 are blocked. :(

**Comment by unclecheese** [comment]
Thanks, guys. Display logic is now 3.2 compatible in the `3.1.*` release.

Sorry for not responding. I forgot to move my MX records and I've been unreachable by Github notifications for six months. :( Lots to catch up on.
